lib/Sema/ConstraintLocator.h

```diff
@@ -715,6 +715,21 @@ class ConstraintLocatorBuilder {
     return (getSummaryFlags() & ConstraintLocator::IsFunctionConversion);
   }
 
+  bool isAutoclosureResult() const {
```

Comment:
I think it make since to rename this to isForAutoClosureResult() just like we have it for isForKeyPathComponent.